### PR TITLE
minio-exporter: remove

### PR DESCRIPTION
**Pull request by msanft**
## Description of changes

Please review this carefully, it is my first module deprecation.

Remove the minio-exporter module, as Minio has built-in Prometheus metrics for a while: https://min.io/docs/minio/linux/operations/monitoring/collect-minio-metrics-using-prometheus.html.

The minio-exporter package doesn't use Go modules, and can't be ported without rewriting of a lot of the source, as it has very outdated dependencies.

Related to #318069

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -903,7 +903,8 @@ mapAliases ({
   nix_2_4 = nixVersions.nix_2_4;
   nix_2_5 = nixVersions.nix_2_5;
   nix_2_6 = nixVersions.nix_2_6;
-  nixfmt = lib.warn "nixfmt was renamed to nixfmt-classic. The nixfmt attribute may be used for the new RFC 166-style formatter in the future, which is currently available as nixfmt-rfc-style" nixfmt-classic; # Added 2024-03-31
+  rian-puppet-go = throw "'librarian-puppet-go' has been removed, as it's upstream is unmaintained"; # Added 2024-06-10
+  libranixfmt = lib.warn "nixfmt was renamed to nixfmt-classic. The nixfmt attribute may be used for the new RFC 166-style formatter in the future, which is currently available as nixfmt-rfc-style" nixfmt-classic; # Added 2024-03-31
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:907]: That seems unintended

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -849,7 +849,7 @@ mapAliases ({
   moonlander = throw "'moonlander' has been removed due to it being broken and unmaintained"; # Added 2023-11-26
   moz-phab = mozphab; # Added 2022-08-09
   mozart-binary = throw "'mozart-binary' has been renamed to/replaced by 'mozart2-binary'"; # Converted to throw 2023-09-10
-  mozart = throw "'mozart' has been renamed to/replaced by 'mozart2-binary'"; # Converted to throw 2023-09-10
+    mozart = throw "'mozart' has been renamed to/replaced by 'mozart2-binary'"; # Converted to throw 2023-09-10
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:852]: this, too

File: nixos/modules/services/monitoring/prometheus/exporters.nix [hunk 3]
```diff
@@ -279,13 +279,19 @@ let
...
                    "jsonExporter" "minioExporter" "nginxExporter" "nodeExporter"
                    "snmpExporter" "unifiExporter" "varnishExporter" ]
-       (opt: lib.mkRemovedOptionModule [ "services" "prometheus" "${opt}" ] ''
-         The prometheus exporters are now configured using `services.prometheus.exporters'.
-         See the 18.03 release notes for more information.
-       '' ));
+      (opt: lib.mkRemovedOptionModule [ "services" "prometheus" "${opt}" ] ''
+        The prometheus exporters are now configured using `services.prometheus.exporters'.
+        See the 18.03 release notes for more information.
+      '' ))
```
Comment (SuperSandro2000) [inline nixos/modules/services/monitoring/prometheus/exporters.nix:289]: Lets drop those very old mkRemovedOptionModule

**Review by bachp — APPROVED** [review]
For me this looks OK.

File: nixos/modules/services/monitoring/prometheus/exporters.nix
Comment (katexochen) [inline nixos/modules/services/monitoring/prometheus/exporters.nix:1]: minio is still in the list of exporterOpts, remove L55.

**Review by katexochen — DISMISSED** [review]

File: nixos/doc/manual/release-notes/rl-2411.section.md [hunk 4]
```diff
@@ -64,6 +64,9 @@
   services.portunus.ldap.package = pkgs.openldap.override { libxcrypt = pkgs.libxcrypt-legacy; };
   ```
 
+- The `services.prometheus.exporters.minio` option has been removed, as it's upstream implementation was broken and unmaintained.
+  Minio now has built-in [Prometheus metrics exposure](https://min.io/docs/minio/linux/operations/monitoring/collect-minio-metrics-using-prometheus.html), which can be used instead.
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply nixos/doc/manual/release-notes/rl-2411.section.md:123]: I don't think that's worth the effort.

**Review by SuperSandro2000 — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**